Mathlib/Analysis/NormedSpace/WithLp.lean

```diff
@@ -0,0 +1,121 @@
+/-
+Copyright (c) 2023 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import Mathlib.Data.Real.ENNReal
+import Mathlib.LinearAlgebra.FiniteDimensional
+
+/-! # The `WithLp` type synonym
+
+`WithLp p V` is the vector space `V`, but with the Lp norm.
```

Comment:
Can you make it clear in the docstring that it is a straight copy of `V`, and therefore that it will only be usable for direct products or finite indexed products (as for infinite products the "right" Lp space is not the same as the direct product of the spaces)?